Server/Pages/Shipment/RegisterShipment.razor

```diff
@@ -0,0 +1,31 @@
+@using Repositories.Entities
+@using Repositories.Interfaces
+
+@code {
+
+    [Inject]
+    public ISenderRepository SenderRepository { get; set; }
+    [Inject]
+    public IPackageRepository PackageRepository { get; set; }
+    [Inject]
+    public IPostMachineRepository PostMachineRepository { get; set; }
+    public Package Package { get; set; }
+    public IEnumerable<PostMachine> PostMachines { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        Package = new Package();
+        await RetrievePostMachines();
+    }
+
+    public async Task RetrievePostMachines()
+    {
+        PostMachines = await PostMachineRepository.GetAsync();
+    }
+
+    public void CreateShipment()
+    {
+        PackageRepository.CreateAsync(Package);
+    }
+
+}
```

Comment:
I'm not familiar with razor, but are there any reasons to be creating these things at all? I have imagined that client side would be making rest calls to controllers for everything. If razor need some kind of code fragments to handle calls, I would prefer for them to be on the client side.

Reply:
Yeah, figured out the template project is run on Blazor Web Assembly. Though it's Blazor Server. Recently I pushed the commit for Blazor Web Assembly, though I still left this piece of code so you can see how it would look like in Blazor Server. What I like about Blazor Server is that it can directly communicate with client and server without controllers. Both the Assembly and the Server has their own pros and cons but I think Blazor Server is more easily programmable. So the example is left for now, we can talk about it in the upcoming sprint session which Blazor approach would fit us the most the most.